File: web3rpc/rpc-specs/code-samples/javascript/klay/block/getNodesByState.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(async () => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");

  // klay_getNodesByState has no dedicated SDK wrapper; call it through the provider.
  const blockNumberOrTag = "latest";
  const states = ["ValActive"]; // omit or pass [] to return nodes in all states
  const result = await provider.send("klay_getNodesByState", [blockNumberOrTag, states]);
  console.log(result);
})();

File: web3rpc/rpc-specs/code-samples/python/klay/block/getNodesByState.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

w3 = Web3(Web3.HTTPProvider(host))

# klay_getNodesByState has no dedicated SDK wrapper; call it through the provider.
# Pass [] (or omit) for the states filter to return nodes in all states.
klay_response = w3.provider.make_request("klay_getNodesByState", ["latest", ["ValActive"]])

print(klay_response)
